### DISCUSSION TITLE
Cannot read property 'resize' of undefined in tsparticles

### DISCUSSION THREAD

**Discussion by ShreyaSheth23** [matteobruni/tsparticles#2091] [Q&A] [answered]
Hii
I am using react-particles-js library for particles animation and it is showing this error for few users.
Here given is the code for particles:
<Particles 
              params={
                {
                      "particles":{
                         "number":{
                            "value":100,
                            "density":{
                               "enable":true,
                               // "value_area":552.4033491425909
                            }
                         },
                         "color":{
                            "value":"random"
                         },
                         "shape":{
                            "type":"circle",
                            "stroke":{
                               "width":0,
                               "color":"#000000"
                            },
                            "polygon":{
                               "nb_sides":3
                            },
                            "image":{
                               "src":"img/github.svg",
                               "width":70,
                               "height":100
                            }
                         },
                         "opacity":{
                            "value":0.2,
                            "random":true,
                            "anim":{
                               "enable":false,
                               "speed":2,
                               "opacity_min":0.2,
                               "sync":false
                            }
                         },
                         "size":{
                            "value":4,
                            "random":true,
                            "anim":{
                               "enable":false,
                               "speed":40,
                               "size_min":0.1,
                               "sync":false
                            }
                         },
                         "line_linked":{
                            "enable":false,
                            "distance":150,
                            "color":"#ffffff",
                            "opacity":0.4,
                            "width":1
                         },
                         "move":{
                            "enable":true,
                            "speed":1.5782952832645452,
                            "direction":"none",
                            "random":true,
                            "straight":false,
                            "out_mode":"out",
                            "bounce":false,
                            "attract":{
                               "enable":false,
                               "rotateX":600,
                               "rotateY":1200
                            }
                         }
                      },
                      "interactivity":{
                         "detect_on":"canvas",
                         "events":{
                            "onhover":{
                               "enable":false,
                               "mode":"repulse"
                            },
                            "onclick":{
                               "enable":true,
                               "mode":"repulse"
                            },
                            "onresize":{
                               "enable":true,
                               "density_auto":true,
                               "density_area":400
                            }
                         },
                         "modes":{
                            "grab":{
                               "distance":400,
                               "line_linked":{
                                  "opacity":1
                               }
                            },
                            "bubble":{
                               "distance":400,
                               "size":40,
                               "duration":2,
                               "opacity":8,
                               "speed":3
                            },
                            "repulse":{
                               "distance":200,
                               "duration":0.4
                            },
                            "push":{
                               "particles_nb":4
                            },
                            "remove":{
                               "particles_nb":2
                            }
                         }
                      },
                      "retina_detect":true
                   }
                }
              height={this.props.parentHeight ? this.props.parentHeight :  '600px'}
              style={{
               borderTop:'1px solid #e2edf6',
               borderBottom:'1px solid #e2edf6',
               boxShadow : 'inset 3px 3px 30px 5px #e2edf6',
                //backgroundImage: `url(${logo})` 
              }}
            />

**Comment by matteobruni** [comment, marked as answer]
Hi @ShreyaSheth23,

I tried your configuration here: https://codesandbox.io/s/busy-grass-l4llm?file=/src/App.js but it works fine.
Can you give more details about this issue, like which browser they are using.

In the sample I've used `react-tsparticles` since it's the official package maintained here. I always recommend the switch since I can fix it quickly if there's a bug.

You can also share the link above with you users, to see if the issue is still there.

**Reply by ShreyaSheth23** [reply]
Yes It is working for me as well but just for few users it is showing this error 
User's configuration:
browserName :Chrome Mobile
browserVersion: 38.0.1025
locale: en-US
osName: Android 4.2
userAgent: Mozilla/5.0 (Linux; Android 4.2.1; en-us; Nexus 5 Build/JOP40D) AppleWebKit/535.19 (KHTML, like Gecko; googleweblight) Chrome/38.0.1025.166 Mobile Safari/535.19

Consoles before error occured:
error
tsParticles error in animation loop
TypeError: Cannot read property 'update' of undefined

warn
error adding particle: TypeError: Cannot read property 'particlePosition' of undefined

**Reply by matteobruni** [reply]
These errors occurs because that browser is not supported by ECMAScript 2015 which is the output of the library. I decided to drop the support of ES5 since it's 6 years old and all major browsers support it, not IE 11 or below but Microsoft is not supporting it anymore. Android 4.2 in 2020 has a 0.8% market share so I'm not gonna fix that.

Supporting ES5 will increase a lot the bundle size, that's why I dropped it in the first place. Anyway you can try adding ES5 poly fills or changing the output of your application using Webpack, Rollup, Parcel, or any other tool. I think one of these solutions could work.

**Reply by ShreyaSheth23** [reply]
So Are you saying to convert our syntax from ES5 to ES6?

**Reply by matteobruni** [reply]
Yes, your output should be ES5 or less to support those devices
